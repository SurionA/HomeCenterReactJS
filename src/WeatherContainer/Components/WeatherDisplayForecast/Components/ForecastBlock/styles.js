export const iconStyle = {
  fontSize: '4rem',
  margin: '15px 0',
};

export const temperatureStyle = {
  fontSize: '1.4rem',
};

export default {
  iconStyle,
  temperatureStyle,
};
