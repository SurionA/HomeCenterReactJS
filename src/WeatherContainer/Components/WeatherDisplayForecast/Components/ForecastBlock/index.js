import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Row } from 'react-bootstrap';

import { iconStyle, temperatureStyle } from './styles';

function ForecastBlock({ weatherData }) {
  return (
    <Fragment>
      <Row>
        <h3 className="title">{weatherData.dayShortName}</h3>
      </Row>
      <Row>
        <div style={iconStyle} className={`wi wi-icon-${weatherData.icon}`} />
      </Row>
      <Row style={temperatureStyle}>{Math.round(parseInt(weatherData.temperature_max, 10))}°</Row>
      <Row style={temperatureStyle}>{Math.round(parseInt(weatherData.temperature_min, 10))}°</Row>
    </Fragment>
  );
}

ForecastBlock.propTypes = {
  weatherData: PropTypes.shape({
    temperature_max: PropTypes.number,
    temperature_min: PropTypes.number,
    icon: PropTypes.number,
    dt: PropTypes.number,
    dayShortName: PropTypes.string,
  }),
};

ForecastBlock.defaultProps = {
  weatherData: PropTypes.shape({}),
};

export default ForecastBlock;
