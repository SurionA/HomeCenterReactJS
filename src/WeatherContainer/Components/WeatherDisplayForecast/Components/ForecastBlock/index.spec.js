import React from 'react';
import renderer from 'react-test-renderer';
import ForecastBlock from './';

it('renders correctly ForecastBlock component', () => {
  const weatherData = {
    temperature_max: 16,
    temperature_min: 12,
    icon: 800,
    dt: 123,
    dayShortName: 'Mon',
  };
  const tree = renderer.create(<ForecastBlock weatherData={weatherData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
