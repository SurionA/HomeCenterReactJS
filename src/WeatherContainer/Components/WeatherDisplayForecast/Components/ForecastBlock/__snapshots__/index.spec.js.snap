// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly ForecastBlock component 1`] = `
Array [
  <div
    className="row"
>
    <h3
        className="title"
    >
        Mon
    </h3>
</div>,
  <div
    className="row"
>
    <div
        className="wi wi-icon-800"
        style={
            Object {
                "fontSize": "4rem",
                "margin": "15px 0",
              }
        }
    />
</div>,
  <div
    className="row"
    style={
        Object {
            "fontSize": "1.4rem",
          }
    }
>
    16
    °
</div>,
  <div
    className="row"
    style={
        Object {
            "fontSize": "1.4rem",
          }
    }
>
    12
    °
</div>,
]
`;
